src/hotspot/cpu/x86/x86.ad

```diff
@@ -8590,6 +8591,20 @@ instruct vpopcountI(vec dst, vec src) %{
   ins_pipe( pipe_slow );
 %}
 
+instruct vpopcountL(vec dst, vec src) %{
+  match(Set dst (PopCountVL src));
+  format %{ "vpopcntq  $dst,$src\t! vector popcount packedL" %}
+  ins_encode %{
+    assert(UsePopCountInstruction, "not enabled");
+
+    int vlen_enc = vector_length_encoding(this, $src);
+    __ vpopcntq($dst$$XMMRegister, $src$$XMMRegister, vlen_enc);
+    __ evpmovqd($dst$$XMMRegister, $dst$$XMMRegister, vlen_enc);
```

Comment:
Hi,
Should this cast be introduced at the middle-end instead? Popcount is a lane-wise operation and forcing the node to do a shape-changing operation seems not so reasonable.
Thanks.